server/src/main/java/org/opensearch/search/aggregations/bucket/adjacency/AdjacencyMatrixAggregationBuilder.java

```diff
@@ -138,13 +146,48 @@ public AdjacencyMatrixAggregationBuilder(String name, String separator, Map<Stri
         setFiltersAsMap(filters);
     }
 
+    /**
+     * @param name
+     *            the name of this aggregation
+     * @param filters
+     *            the filters and their key to use with this aggregation.
+     * @param showOnlyIntersecting
+     *            show only the buckets that intersection multiple documents
+     */
+    public AdjacencyMatrixAggregationBuilder(String name, Map<String, QueryBuilder> filters, boolean showOnlyIntersecting) {
+        this(name, DEFAULT_SEPARATOR, filters, showOnlyIntersecting);
+    }
+
+    /**
+     * @param name
+     *            the name of this aggregation
+     * @param separator
+     *            the string used to separate keys in intersections buckets e.g.
+     *            &amp; character for keyed filters A and B would return an
+     *            intersection bucket named A&amp;B
+     * @param filters
+     *            the filters and their key to use with this aggregation.
+     * @param showOnlyIntersecting
+     *            show only the buckets that intersection multiple documents
+     */
+    public AdjacencyMatrixAggregationBuilder(
+        String name,
+        String separator,
+        Map<String, QueryBuilder> filters,
+        boolean showOnlyIntersecting
+    ) {
+        this(name, separator, filters);
+        this.showOnlyIntersecting = showOnlyIntersecting;
+    }
+
     /**
      * Read from a stream.
      */
     public AdjacencyMatrixAggregationBuilder(StreamInput in) throws IOException {
         super(in);
         int filtersSize = in.readVInt();
         separator = in.readString();
+        showOnlyIntersecting = in.readBoolean();
```

Comment:
Unfortunately, when we add a new field to a serialized object, we need to go through a painful dance to manage backwards compatibility:

1. Make the change on `main`, wrapping the serialization/deserialization behind a check like `if (in.getVersion().onOrAfter(Version.V_3_0_0)`.
2. When we try to backport the change to 2.x, the backport will fail to build, because the 2.x branch doesn't know about `V_3_0_0`. So, you would need to add a commit to update the check to `V_2_18_0` (or whatever the next release will be).
3. Finally, to avoid breaking the build on `main`, we need to update the check to say `V_2_18_0` there too.